### PR TITLE
Blog comments

### PR DESCRIPTION
**Pull request by anushadatta**
Added a comments section to blog page. As of now, JavaScript used to retrieve hard coded comments from server with fetch(). Consequently, those comments are added to the DOM.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java [hunk 1]
```diff
@@ -14,19 +14,40 @@
...
+@WebServlet("/blog-comments")
+public final class BlogCommentsServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    // Comments data
+    Date postDate = new Date();
+    String comment = "That's a wonderful blog!";
```
Comment (rashikaranpuria) [inline portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java:33]: Consider removing the temporary logic to another function so it's easy to see what this method will look like finally.

File: portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java [hunk 2]
```diff
@@ -14,19 +14,40 @@
...
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  /**
+   * Converts a BlogComments instance into a JSON string using the Gson library. Note: We first added
+   * the Gson library dependency to pom.xml.
+   */
+  private String convertToJsonUsingGson(BlogComments commentsData) {
+    Gson gson = new Gson();
```
Comment (rashikaranpuria) [inline portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java:49]: If we are going to call this method multiple times then we may initialize Gson separately only once

File: portfolio/src/main/java/com/google/sps/data/BlogComments.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package com.google.sps.data;
+
+import java.util.Date;
+
+/** Class containing blog comments. */
+public final class BlogComments {
+
```
[the remark this answers is not in the document]
Comment (anushadatta) [reply portfolio/src/main/java/com/google/sps/data/BlogComments.java:7]: It is supposed to a be a single comment. So for more clarity, should I change the class name to **BlogComment** and the **comment** attribute name to **text**?

File: portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java [hunk 4]
```diff
@@ -14,19 +14,40 @@
...
+import java.util.Date;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
-public class DataServlet extends HttpServlet {
+@WebServlet("/blog-comments")
```
[the remark this answers is not in the document]
Comment (anushadatta) [reply portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java:26]: Yes alright, I'll change that to **/comments**

File: portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java [hunk 5]
```diff
@@ -14,19 +14,40 @@
...
+    // Comments data
+    Date postDate = new Date();
+    String comment = "That's a wonderful blog!";
+
+    // Convert the comments data to JSON
+    BlogComments commentsData = new BlogComments(postDate, comment);
+    String json = convertToJsonUsingGson(commentsData);
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
```
[the remark this answers is not in the document]
Comment (anushadatta) [reply portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java:40]: Oh no, the semi colon is my mistake, I'll remove that. How do I define the content type as a constant field?

File: portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java (hunk 1 above)
Comment (anushadatta) [reply portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java:33]: Okay that's a great insight, I'll keep that in mind for my next PRs.

File: portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java (hunk 2 above)
Comment (anushadatta) [reply portfolio/src/main/java/com/google/sps/servlets/BlogCommentsServlet.java:49]: Okay, so should I initialize it in the beginning of the class outside the function? I can then pass the **gson** object instance as a parameter to this function.

**Comment by anushadatta** [comment]
Okay thanks, @rashikaranpuria could you also please review my most recent merge also? Then I'll be sorted for all this week's deliverables.

File: portfolio/src/main/java/com/google/sps/servlets/LoadCommentsServlet.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    Query query = new Query("BlogComment").addSort("postDate", sortingOrder);
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<BlogComment> commentsList = new ArrayList<BlogComment>();
+
+    for (Entity entity : results.asIterable()) {
+
+      Date postDate = (Date) entity.getProperty("postDate");
```
Comment (rashikaranpuria) [inline portfolio/src/main/java/com/google/sps/servlets/LoadCommentsServlet.java:64]: When using strings such as postDate, text, BlogComment to get properties of the entity or entity itself like above, I feel that maybe we can define these as constants. Maybe in The Entity class itself like BlogComment in this case

File: portfolio/src/main/java/com/google/sps/servlets/LoadCommentsServlet.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+      String text = (String) entity.getProperty("text");
+
+      BlogComment c = new BlogComment(postDate, text);
```
Comment (rashikaranpuria) [inline portfolio/src/main/java/com/google/sps/servlets/LoadCommentsServlet.java:67]: Could be a better variable name than c. SIngle character names promote bad readability.

**Comment by rashikaranpuria** [comment]
Everything looks good, just some optional comments. Feel free to merge without further review.

**Pull request merged**